### PR TITLE
TCP Keep Alive Control

### PR DESCRIPTION
**Pull request by xiaokangwang**
This is a reworked PR of https://github.com/v2fly/v2ray-core/pull/1276.

### CODE REVIEW

File: transport/internet/sockopt_darwin.go [hunk 1]
```diff
@@ -10,7 +10,8 @@ const (
 	// TCP_FASTOPEN_SERVER is the value to enable TCP fast open on darwin for server connections.
 	TCP_FASTOPEN_SERVER = 0x01 // nolint: golint,stylecheck
 	// TCP_FASTOPEN_CLIENT is the value to enable TCP fast open on darwin for client connections.
-	TCP_FASTOPEN_CLIENT = 0x02 // nolint: golint,stylecheck
+	TCP_FASTOPEN_CLIENT = 0x02  // nolint: golint,stylecheck
+	TCP_KEEPINTVL       = 0x101 // nolint: golint,stylecheck
```
Comment (database64128) [inline transport/internet/sockopt_darwin.go:14]: This is not necessary, see https://pkg.go.dev/golang.org/x/sys/unix?GOOS=darwin.

**Comment by database64128** [comment]
The `syscall` package recommends using `golang.org/x/sys/unix` and `golang.org/x/sys/windows` instead of `syscall`.

> Deprecated: this package is locked down. Callers should use the corresponding package in the golang.org/x/sys repository instead. That is also where updates required by new systems or versions should be applied. See https://golang.org/s/go1.4-syscall for more information.

**Comment by xiaokangwang** [comment]
Okay, I will rewrite these code to use unix instead of syscall.

**Comment by ValdikSS** [comment]
Unfortunately this patchset does not handle keep-alive on `accept()`'ed sockets. Regardless of configured value, it's always 15 seconds.

**Comment by ValdikSS** [comment]
Oh wait, the patch hasn't been backported to v4.

**Pull request merged**